utils/build-script

```diff
@@ -1338,10 +1289,61 @@ details of the setups of other systems or automated environments.""")
             "--install-symroot", os.path.abspath(args.install_symroot)
         ]
 
+    # Create the build target list.
+    #
+    # Take all of the configured stdlib-deployment-targets
+    # and subtract any we've been told to skip.
+
+    skipped_targets = []
+    if args.skip_build_linux:
+        skipped_targets += StdlibDeploymentTarget.Linux.allArchs
+    if args.skip_build_freebsd:
+        skipped_targets += StdlibDeploymentTarget.FreeBSD.allArchs
+    if args.skip_build_cygwin:
+        skipped_targets += StdlibDeploymentTarget.Cygwin.allArchs
+    if args.skip_build_android:
+        skipped_targets += StdlibDeploymentTarget.Android.allArchs
+    if args.skip_build_osx:
+        skipped_targets += StdlibDeploymentTarget.OSX.allArchs
+
+    # iOS targets are special: they are configured by default, but only built
+    # if the appropriate '--ios'/'--tvos'/... flag is given.
+    if args.skip_build_ios_device or not args.ios:
+        skipped_targets += StdlibDeploymentTarget.iOS.allArchs
+    if args.skip_build_ios_simulator or not args.ios:
+        skipped_targets += StdlibDeploymentTarget.iOSSimulator.allArchs
+    if args.skip_build_tvos_device or not args.tvos:
+        skipped_targets += StdlibDeploymentTarget.AppleTV.allArchs
+    if args.skip_build_tvos_simulator or not args.tvos:
+        skipped_targets += StdlibDeploymentTarget.AppleTVSimulator.allArchs
+    if args.skip_build_watchos_device or not args.watchos:
+        skipped_targets += StdlibDeploymentTarget.AppleWatch.allArchs
+    if args.skip_build_watchos_simulator or not args.watchos:
+        skipped_targets += StdlibDeploymentTarget.AppleWatchSimulator.allArchs
+
+    build_stdlib_deployment_targets = [
+        x for x in args.stdlib_deployment_targets
+        if x not in skipped_targets]
+
+    if args.skip_build or len(build_stdlib_deployment_targets) == 0:
+        build_stdlib_deployment_targets = ['none']
```

Comment:
Please consider to factor out the calculation of
`build_stdlib_deployment_targets` and `test_deployment_targets` into
`swift_build_support` package and write some tests.